qlib/workflow/__init__.py

```diff
@@ -370,11 +369,11 @@ def uri_context(self, uri: Text):
             the temporal uri
         """
         prev_uri = self.exp_manager.default_uri
-        C.exp_manager["kwargs"]["uri"] = uri
+        self.exp_manager.default_uri = uri
```

Comment:
I think the set of default_uri is not engouh. The self.exp_manager is not change the client uri if you only change default uri.

Reply:
It is fixed now.
The _client will not be cached.  No further maintenance is required for it.